salt/modules/status.py

```diff
@@ -148,26 +148,42 @@ def uptime():
 
         salt '*' status.uptime
     '''
-    ut_ret = {'seconds': 0}
+    curr_seconds = time.time()
+
+    # Get uptime in seconds
     if salt.utils.is_linux():
         ut_path = "/proc/uptime"
         if not os.path.exists(ut_path):
             raise CommandExecutionError("File {ut_path} was not found.".format(ut_path=ut_path))
-        ut_ret['seconds'] = int(float(open(ut_path).read().strip().split()[0]))
+        seconds = int(float(salt.utils.fopen(ut_path).read().split()[0]))
     elif salt.utils.is_sunos():
         cmd = "kstat -p unix:0:system_misc:boot_time | nawk '{printf \"%d\\n\", srand()-$2}'"
-        ut_ret['seconds'] = int(__salt__['cmd.shell'](cmd, output_loglevel='trace').strip() or 0)
+        seconds = int(float(__salt__['cmd.shell'](cmd, output_loglevel='trace').strip() or 0))
+    elif salt.utils.is_darwin():
+        bt_data = __salt__['sysctl.get']('kern.boottime')
+        if not bt_data:
+            raise CommandExecutionError('Cannot find kern.boottime system parameter')
+        sec_data, usec_data = bt_data.split('}')[0].strip(' {').split(', ')
+        seconds = int(curr_seconds - int(sec_data.split('sec = ')[1]))
```

Comment:
Specifically:
```python
data = bt_data.split("{")[-1].split("}")[0].strip().replace(' ', '')
uptime = dict([(k_sec, int(v_sec,))
                for k_sec, v_sec in [pair.strip().split("=")
                                     for pair in data.split(",")]])
```
What you do here is to turn mac returned data into a dictionary, and refer to it something like:
```python
uptime['usec']
uptime['sec']
```

You can turn the above to a lambda and also query sleep time etc.